lib/ProductOpener/KnowledgePanels.pm

```diff
@@ -106,6 +106,19 @@ sub initialize_knowledge_panels_options ($knowledge_panels_options_ref, $request
 	}
 	$knowledge_panels_options_ref->{knowledge_panels_client} = $knowledge_panels_client;
 
+	my $included_panels = single_param('knowledge_panels_included') || '';
+	my %included_panels = map {$_ => 1} split(/,/, $included_panels);
+	my $excluded_panels = single_param('knowledge_panels_excluded') || '';
+	# excluded overrides included
+	$included_panels{$_} = 0 for split(/,/, $excluded_panels);
+	$knowledge_panels_options_ref->{knowledge_panels_includes} = sub {
+		my $panel_id = shift;
+		return (
+				   (exists $included_panels{$panel_id}) and ($included_panels{$panel_id} eq 1)
+				or (not(exists $included_panels{$panel_id}) and not $included_panels)
```

Comment:
The condition is not that easy to read.
If we use undef above why not:
```suggestion
					not $included_panels or (defined $included_panels{$panel_id})
```

Reply:
It didn't work, in facts its the 'exists' that is really important. Finally I did this, but if you see a better way ... 
https://github.com/openfoodfacts/openfoodfacts-server/blob/3ec9ae8bb5cdad82277b581f21becee500016b5a/lib/ProductOpener/KnowledgePanels.pm#L116-L117